src/components/pages/ClassInfoPage.jsx

```diff
@@ -0,0 +1,227 @@
+import React, { useEffect, useState } from 'react';
```

Comment:
There is a lot going on in this component. Let's separate concerns into `TeachInfo.jsx` and ClassInfoPage.jsx`